### ISSUE TITLE
Extra '0' param being added when composing prefixed routes

### ISSUE THREAD

**Issue by lourd** [alexmingoia/koa-router#247]
I ran across this bug when composing resource-specific routers with a top-level router.
## tl;dr

There's an extra key of `'0'` being added onto `ctx.params` when composing prefixed routers
## Versions
- koa@2.0.0-alpha.3
- koa-router@7.0.1
- node@4.2.1
## Reproduction

``` sh
$ npm install koa@next koa-router@7
```
### Case 1

``` js
const Koa = require('koa')
const app = new Koa()
const Router = require('koa-router')

const topLevelRouter = new Router({
  prefix: '/v1'
})

const secondaryRouter = new Router({
  prefix: '/things'
})

secondaryRouter.get('/',  (ctx) =>  {
    console.log(ctx.params)
    ctx.body = 'hello world'
})

topLevelRouter.use(secondaryRouter.routes())

app.use(topLevelRouter.routes())
app.listen(3000)
```

``` sh
$ curl 0.0.0.0:3000/v1/things
# client log
> hello world
# server log
> { '0': '' }
```
### Case 2

I discovered that giving an empty string as an argument to `topLevelRouter.use` got rid of the extra parameter

``` js
const Koa = require('koa')
const app = new Koa()
const Router = require('koa-router')

const topLevelRouter = new Router({
  prefix: '/v1'
})

const secondaryRouter = new Router({
  prefix: '/things'
})

secondaryRouter.get('/',  (ctx) =>  {
    console.log(ctx.params)
    ctx.body = 'hello world'
})

// NOTE the empty string as the first argument
topLevelRouter.use('', secondaryRouter.routes())

app.use(topLevelRouter.routes())
app.listen(3000)
```

``` sh
$ curl 0.0.0.0:3000/v1/things
# Client log
> hello world
# server log
> {}
```
### Case 3

Putting the prefix in `topLevelRouter.use` also doesn't include the extra parameter

``` js
const Koa = require('koa')
const app = new Koa()
const Router = require('koa-router')

const topLevelRouter = new Router({
  prefix: '/v1'
})

const secondaryRouter = new Router()

secondaryRouter.get('/',  (ctx) =>  {
    console.log(ctx.params)
    ctx.body = 'hello world'
})

topLevelRouter.use('/things', secondaryRouter.routes())

app.use(topLevelRouter.routes())
app.listen(3000)
```

``` sh
$ curl 0.0.0.0:3000/v1/things
# Client log
> hello world
# Server log
> {}
```

**Comment by aheckmann-pebble** [comment]
yeah just ran into this too. here's a simple way to reproduce:

```js
let router = new Router();
router.use(someMiddleware()); // this is the trigger
router.get('/', function*(){
  console.log(this.params); //--> { "0": "/" }
});

///

let router = new Router();
router.get('/', function*(){
  console.log(this.params); //--> { }
});
```

**Comment by acepukas** [comment]
This was bugging me too. I found the culprit. In the `Router.use` function, if no path is supplied as first argument, it will default to `(.*)` which the path-to-regexp lib interprets as "capture everything".
path-to-regexp gives the capture a numeric index as opposed to a string name, like in the case of `/foo/:foo`. That's why it appears in the `ctx.params` object as `{ '0': '/the/entire/path' }`.

@lourd That's why supplying an empty path to `Router.use` suppresses the issue. I manually changed the default path to be an empty string in my dev environment and that seems to resolve the issue though I can't say if that would have an adverse effect elsewhere.

koa-router by design seems to place any captures into the ctx.params object, which makes sense, though I think specifying `(.*)` as the default path might have been a mistake?
